statsmodels/base/distributed_estimation.py

```diff
@@ -0,0 +1,410 @@
+        machine specific sample size.
+    alpha : scalar or array-like
+        The penalty weight.  If a scalar, the same penalty weight
+        applies to all variables in the model.  If a vector, it
+        must have the same length as `params`, and contains a
+        penalty weight for each coefficient.
+    L1_wt : scalar
+        The fraction of the penalty given to the L1 penalty term.
+        Must be between 0 and 1 (inclusive).  If 0, the fit is
+        a ridge fit, if 1 it is a lasso fit.
+    score_kwds : dict-like or None
+        Keyword arguments for the score function.
+    
+    Returns
+    -------
+    An array-like object of the same dimension as beta_hat
+   
+    Notes
+    -----
+    In general:
+    
+    nabla l_k(beta)
+
+    For the simple linear case:
+
+    X^T(y - X^T beta) / n
+    """
+
+    grad = -mod.score(np.r_[beta_hat], **score_kwds) / n
+    # this second part comes from the elastic net penalty
+    grad += alpha * (1 - L1_wt)
+    return grad
+
+
+def _gen_wdesign_mat(mod, beta_hat, hess_kwds):
+    """
+    generates the weighted design matrix necessary to generate
+    the approximate inverse covariance matrix
+
+    Parameters
+    ----------
+    mod : statsmodels model class
+        The model for the current machine.
+    beta_hat : array-like
+        The estimated coefficients for the current machine.
+    hess_kwds : dict-like or None
+        Keyword arguments for the hessian function.
+
+    Returns
+    -------
+    An array-like object, updated design matrix, same dimension
+    as mod.exog
+    """
+
+#    X_beta = np.sqrt(mod.hessian_obs(np.r_[beta_hat], **hess_kwds)).dot(mod.exog)
+#    X_beta = X_beta - X_beta.mean(0) / X_beta.std(0)
+#    return X_beta
+    return np.sqrt(mod.hessian_obs(np.r_[beta_hat], **hess_kwds)).dot(mod.exog)
+#    return mod.exog
+    # TODO need to handle duration and other linear model classes
+
+
+def _gen_gamma_hat(X_beta, pi, p, n, alpha):
+    """
+    generates the gamma hat values for the pith variable, used to
+    estimate theta hat.
+
+    Parameters
+    ----------
+    X_beta : array-like
+        The weighted design matrix for the current machine
+    pi : scalar
+        Index of the current variable
+    p : scalar
+        Number of variables
+    n : scalar
+        sample size of current machine
+    alpha : scalar or array-like
+        The penalty weight.  If a scalar, the same penalty weight
+        applies to all variables in the model.  If a vector, it
+        must have the same length as `params`, and contains a
+        penalty weight for each coefficient.
+
+    Returns
+    -------
+    An array-like object of length p-1
+
+    Notes
+    -----
+
+    gamma_hat_i = arg min 1/(2n) ||X_beta,i - X_beta,-i gamma||_2^2
+                          + alpha ||gamma||_1
+    """
+   
+    ind = range(pi) + range(pi + 1, p)
+
+    # TODO use elastic net optimization routine directly
+    tmod = OLS(X_beta[:, pi], X_beta[:, ind])
```

Comment:
Also, the model_class for distributed_estimation currently defaults to OLS, though the handling of this could be changed to only import OLS if model_class was None in the case where gamma calls the elastic_net code directly.